### PR TITLE
Add img2img stable diffusion

### PR DESCRIPTION
**Pull request by jinchen62**

### CODE REVIEW

File: shark/iree_utils/vulkan_utils.py [hunk 1]
```diff
@@ -78,6 +78,8 @@ def get_vulkan_target_triple(device_name):
         triple = f"turing-rtx6000-{system_os}"
     elif all(x in device_name for x in ("RTX", "8000")):
         triple = f"turing-rtx8000-{system_os}"
+    elif all(x in device_name for x in ("TITAN", "RTX")):
+        triple = f"turing-titanrtx-{system_os}"
```
Comment (powderluv) [inline shark/iree_utils/vulkan_utils.py:82]: Always a good idea to add this separately

**Review by powderluv — COMMENTED** [review]
Does main.py work via the cli too ?

**Comment by Shukla-Gaurav** [comment]
@jinchen62 The img2img should be a different tab(there will be a different file for it's web UI, not to be mixed with txt2img). Also the encode part should be going through the shark pipeline.

File: web/models/stable_diffusion/main.py [hunk 2]
```diff
@@ -58,16 +60,40 @@ def set_ui_params(
...
+        subfolder="vae",
+    )
+    vae = vae.to("cuda")
+
+    img_arr = np.stack([np.array(img) for img in imgs], axis=0)
+    img_arr = img_arr / 255.0
+    img_arr = torch.from_numpy(img_arr).half().permute(0, 3, 1, 2)
+    img_arr = 2 * (img_arr - 0.5)
+
+    latent_dists = vae.encode(img_arr.to("cuda"))["latent_dist"]
```
Comment (Shukla-Gaurav) [inline web/models/stable_diffusion/main.py:87]: This should be going through shark pipeline exactly like vae.decode.

**Review by Shukla-Gaurav — CHANGES_REQUESTED** [review]

**Comment by fokin33** [comment]
if no img2img - it ok:
(shark.venv) PS H:\SHARK\web> python index.py --max_length=77 --vulkan_large_heap_block_size=0 --use_base_vae --local_tank_cache h:\shark\TEMP
vulkan devices are available.
cuda devices are not available.
Running on local URL:  http://0.0.0.0:8080

To create a public link, set `share=True` in `launch()`.
Found device AMD Radeon RX 580 2048SP. Using target triple rdna2-unknown-windows.
Tuned models are currently not supported for this setting.
Using h:\shark\TEMP as local shark_tank cache directory.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 1.25kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 29.1kB/s]
loading existing vmfb from: H:\SHARK\web\euler_scale_model_input_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 1.28kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 32.0kB/s]
loading existing vmfb from: H:\SHARK\web\euler_step_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 1.26kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 33.7kB/s]
loading existing vmfb from: H:\SHARK\web\vae2base_8dec_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 1.17kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 32.0kB/s]
loading existing vmfb from: H:\SHARK\web\unet_19dec_v2p1base_fp16_64_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 1.23kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 30.5kB/s]
loading existing vmfb from: H:\SHARK\web\clip_19dec_v2p1base_fp32_64_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
2it [00:02,  1.15s/it]

Average step time: 1146.44193649292ms/it
Clip Inference time (ms) = 87.972
VAE Inference time (ms): 1618.157

Total image generation time: 4.1299662590026855sec


If img2img - error below:
Cannot initialize model with low cpu memory usage because `accelerate` was not found in the environment. Defaulting to `low_cpu_mem_usage=False`. It is strongly recommended to install `accelerate` for faster and less memory-intense model loading. You can do so with:
```
pip install accelerate
```
.
Traceback (most recent call last):
  File "H:\SHARK\shark.venv\lib\site-packages\gradio\routes.py", line 321, in run_predict
    output = await app.blocks.process_api(
  File "H:\SHARK\shark.venv\lib\site-packages\gradio\blocks.py", line 1015, in process_api
    result = await self.call_function(fn_index, inputs, iterator, request)
  File "H:\SHARK\shark.venv\lib\site-packages\gradio\blocks.py", line 856, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "H:\SHARK\shark.venv\lib\site-packages\anyio\to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "H:\SHARK\shark.venv\lib\site-packages\anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "H:\SHARK\shark.venv\lib\site-packages\anyio\_backends\_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "H:\SHARK\web\models\stable_diffusion\main.py", line 148, in stable_diff_inf
    img_latents = encode_img_latents(args, init_img)
  File "H:\SHARK\web\models\stable_diffusion\main.py", line 80, in encode_img_latents
    vae = vae.to("cuda")
  File "H:\SHARK\shark.venv\lib\site-packages\torch\nn\modules\module.py", line 1132, in to
    return self._apply(convert)
  File "H:\SHARK\shark.venv\lib\site-packages\torch\nn\modules\module.py", line 784, in _apply
    module._apply(fn)
  File "H:\SHARK\shark.venv\lib\site-packages\torch\nn\modules\module.py", line 784, in _apply
    module._apply(fn)
  File "H:\SHARK\shark.venv\lib\site-packages\torch\nn\modules\module.py", line 807, in _apply
    param_applied = fn(param)
  File "H:\SHARK\shark.venv\lib\site-packages\torch\nn\modules\module.py", line 1130, in convert
    return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
  File "H:\SHARK\shark.venv\lib\site-packages\torch\cuda\__init__.py", line 232, in _lazy_init
    raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled

**Comment by jinchen62** [comment]
@fokin33 Hi, you need to install torch with cuda.
`pip install --pre torch torchvision --extra-index-url https://download.pytorch.org/whl/nightly/cu117`

**Comment by fokin33** [comment]
I am running on AMD.
(shark.venv) PS H:\SHARK\web> pip install --pre torch torchvision --extra-index-url https://download.pytorch.org/whl/nightly/cu117
Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/nightly/cu117
Requirement already satisfied: torch in h:\shark\shark.venv\lib\site-packages (2.0.0.dev20221221+cpu)
Requirement already satisfied: torchvision in h:\shark\shark.venv\lib\site-packages (0.15.0.dev20221221+cpu)
Requirement already satisfied: sympy in h:\shark\shark.venv\lib\site-packages (from torch) (1.11.1)
Requirement already satisfied: networkx in h:\shark\shark.venv\lib\site-packages (from torch) (3.0rc1)
Requirement already satisfied: typing-extensions in h:\shark\shark.venv\lib\site-packages (from torch) (4.4.0)
Requirement already satisfied: numpy in h:\shark\shark.venv\lib\site-packages (from torchvision) (1.24.0)
Requirement already satisfied: requests in h:\shark\shark.venv\lib\site-packages (from torchvision) (2.28.1)
Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in h:\shark\shark.venv\lib\site-packages (from torchvision) (9.3.0)
Requirement already satisfied: charset-normalizer<3,>=2 in h:\shark\shark.venv\lib\site-packages (from requests->torchvision) (2.1.1)
Requirement already satisfied: idna<4,>=2.5 in h:\shark\shark.venv\lib\site-packages (from requests->torchvision) (3.4)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in h:\shark\shark.venv\lib\site-packages (from requests->torchvision) (1.26.13)
Requirement already satisfied: certifi>=2017.4.17 in h:\shark\shark.venv\lib\site-packages (from requests->torchvision) (2022.12.7)
Requirement already satisfied: mpmath>=0.19 in h:\shark\shark.venv\lib\site-packages (from sympy->torch) (1.2.1)
(shark.venv) PS H:\SHARK\web> python index.py --max_length=77 --vulkan_large_heap_block_size=0 --use_base_vae --local_tank_cache h:\shark\TEMP
vulkan devices are available.
cuda devices are not available.
Running on local URL:  http://0.0.0.0:8080

To create a public link, set `share=True` in `launch()`.
Found device AMD Radeon RX 580 2048SP. Using target triple rdna2-unknown-windows.
Tuned models are currently not supported for this setting.
Using h:\shark\TEMP as local shark_tank cache directory.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 32.0kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 33.7kB/s]
loading existing vmfb from: H:\SHARK\web\euler_scale_model_input_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 27.8kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 29.1kB/s]
loading existing vmfb from: H:\SHARK\web\euler_step_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 37.7kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 37.7kB/s]
loading existing vmfb from: H:\SHARK\web\vae2base_8dec_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 32.0kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 33.7kB/s]
loading existing vmfb from: H:\SHARK\web\unet_19dec_v2p1base_fp16_64_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 27.8kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 29.1kB/s]
loading existing vmfb from: H:\SHARK\web\clip_19dec_v2p1base_fp32_64_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Cannot initialize model with low cpu memory usage because `accelerate` was not found in the environment. Defaulting to `low_cpu_mem_usage=False`. It is strongly recommended to install `accelerate` for faster and less memory-intense model loading. You can do so with:
```
pip install accelerate
```
.
Traceback (most recent call last):
  File "H:\SHARK\shark.venv\lib\site-packages\gradio\routes.py", line 321, in run_predict
    output = await app.blocks.process_api(
  File "H:\SHARK\shark.venv\lib\site-packages\gradio\blocks.py", line 1015, in process_api
    result = await self.call_function(fn_index, inputs, iterator, request)
  File "H:\SHARK\shark.venv\lib\site-packages\gradio\blocks.py", line 856, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "H:\SHARK\shark.venv\lib\site-packages\anyio\to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "H:\SHARK\shark.venv\lib\site-packages\anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "H:\SHARK\shark.venv\lib\site-packages\anyio\_backends\_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "H:\SHARK\web\models\stable_diffusion\main.py", line 148, in stable_diff_inf
    img_latents = encode_img_latents(args, init_img)
  File "H:\SHARK\web\models\stable_diffusion\main.py", line 80, in encode_img_latents
    vae = vae.to("cuda")
  File "H:\SHARK\shark.venv\lib\site-packages\torch\nn\modules\module.py", line 1132, in to
    return self._apply(convert)
  File "H:\SHARK\shark.venv\lib\site-packages\torch\nn\modules\module.py", line 784, in _apply
    module._apply(fn)
  File "H:\SHARK\shark.venv\lib\site-packages\torch\nn\modules\module.py", line 784, in _apply
    module._apply(fn)
  File "H:\SHARK\shark.venv\lib\site-packages\torch\nn\modules\module.py", line 807, in _apply
    param_applied = fn(param)
  File "H:\SHARK\shark.venv\lib\site-packages\torch\nn\modules\module.py", line 1130, in convert
    return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
  File "H:\SHARK\shark.venv\lib\site-packages\torch\cuda\__init__.py", line 232, in _lazy_init
    raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled

**Comment by fokin33** [comment]
(shark.venv) PS H:\shark\web> git fetch origin
(shark.venv) PS H:\shark\web> git reset --hard origin/main
HEAD is now at 12f844d Git pull through argument in setup_venv (#623)
(shark.venv) PS H:\shark\web> git pull origin pull/753/head
remote: Enumerating objects: 23, done.
remote: Counting objects: 100% (23/23), done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 23 (delta 16), reused 23 (delta 16), pack-reused 0
Unpacking objects: 100% (23/23), 7.46 KiB | 173.00 KiB/s, done.
From https://github.com/nod-ai/SHARK
 * branch            refs/pull/753/head -> FETCH_HEAD
Updating 12f844d..82d2998
Fast-forward
 .../shark_inference/stable_diffusion/main.py       |  12 +-
 .../stable_diffusion/model_wrappers.py             |  65 ++++++
 .../shark_inference/stable_diffusion/opt_params.py |  17 ++
 .../stable_diffusion/resources/model_db.json       |   3 +
 .../stable_diffusion/stable_args.py                |   6 +
 web/img2img.py                                     | 171 ++++++++++++++
 web/models/stable_diffusion/cache_objects.py       |   8 +-
 web/models/stable_diffusion/img2img_main.py        | 246 +++++++++++++++++++++
 web/models/stable_diffusion/model_wrappers.py      |  29 +++
 web/models/stable_diffusion/opt_params.py          |  15 ++
 .../stable_diffusion/resources/model_db.json       |   3 +
 web/models/stable_diffusion/stable_args.py         |   6 +
 12 files changed, 579 insertions(+), 2 deletions(-)
 create mode 100644 web/img2img.py
 create mode 100644 web/models/stable_diffusion/img2img_main.py
(shark.venv) PS H:\shark\web> python img2img.py --max_length=77 --vulkan_large_heap_block_size=0 --use_base_vae --local_tank_cache h:\shark\TEMP
vulkan devices are available.
cuda devices are not available.
Running on local URL:  http://0.0.0.0:8080

To create a public link, set `share=True` in `launch()`.
Found device AMD Radeon RX 580 2048SP. Using target triple rdna2-unknown-windows.
Tuned models are currently not supported for this setting.
Using h:\shark\TEMP as local shark_tank cache directory.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 1.22kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 27.8kB/s]
loading existing vmfb from: H:\shark\web\euler_scale_model_input_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 1.18kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 25.6kB/s]
loading existing vmfb from: H:\shark\web\euler_step_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Traceback (most recent call last):
  File "H:\shark\web\models\stable_diffusion\opt_params.py", line 31, in get_params
    model_name = models_db[1][model_key]
KeyError: 'stablediffusion/v2_1base/vae_encode/fp16/length_77/untuned/base'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "H:\shark\shark.venv\lib\site-packages\gradio\routes.py", line 321, in run_predict
    output = await app.blocks.process_api(
  File "H:\shark\shark.venv\lib\site-packages\gradio\blocks.py", line 1015, in process_api
    result = await self.call_function(fn_index, inputs, iterator, request)
  File "H:\shark\shark.venv\lib\site-packages\gradio\blocks.py", line 856, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "H:\shark\shark.venv\lib\site-packages\anyio\to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "H:\shark\shark.venv\lib\site-packages\anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "H:\shark\shark.venv\lib\site-packages\anyio\_backends\_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "H:\shark\web\models\stable_diffusion\img2img_main.py", line 111, in img2img_inf
    model_cache.set_models(device_key)
  File "H:\shark\web\models\stable_diffusion\cache_objects.py", line 111, in set_models
    self.vae_encode = get_vae_encode()
  File "H:\shark\web\models\stable_diffusion\opt_params.py", line 82, in get_vae_encode
    bucket, model_name, iree_flags = get_params(
  File "H:\shark\web\models\stable_diffusion\opt_params.py", line 36, in get_params
    raise Exception(
Exception:  there is no entry for stablediffusion/v2_1base/vae_encode/fp16/length_77/untuned/base in the models databaseKeyboard

**Comment by jinchen62** [comment]
@fokin33 fixed the model path issue, thanks for pointing it out

**Comment by fokin33** [comment]
Unfortunately, now there's another problem:
(shark.venv) PS H:\branches\shark\web> python img2img.py --max_length=77 --vulkan_large_heap_block_size=0 --use_base_vae --local_tank_cache h:\shark\TEMP
vulkan devices are available.
cuda devices are not available.
Running on local URL:  http://0.0.0.0:8080

To create a public link, set `share=True` in `launch()`.
Found device AMD Radeon RX 580 2048SP. Using target triple rdna2-unknown-windows.
Tuned models are currently not supported for this setting.
Using h:\shark\TEMP as local shark_tank cache directory.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 1.29kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 22.1kB/s]
loading existing vmfb from: H:\branches\shark\web\euler_scale_model_input_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 1.25kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 22.1kB/s]
loading existing vmfb from: H:\branches\shark\web\euler_step_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Downloading artifacts for model vae2base_encode_9jan_fp16...
100%|███████████████████████████████████████████████████████████████████████████████████| 156/156 [00:00<00:00, 284B/s]
100%|██████████████████████████████████████████████████████████████████████████████| 32.3k/32.3k [00:00<00:00, 155kB/s]
100%|█████████████████████████████████████████████████████████████████████████████| 1.50M/1.50M [00:01<00:00, 1.32MB/s]
100%|█████████████████████████████████████████████████████████████████████████████| 66.7M/66.7M [00:07<00:00, 9.82MB/s]
loading existing vmfb from: H:\branches\shark\web\vae2base_encode_9jan_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 1.04kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 26.7kB/s]
loading existing vmfb from: H:\branches\shark\web\vae2base_8dec_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 1.24kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 24.6kB/s]
loading existing vmfb from: H:\branches\shark\web\unet_19dec_v2p1base_fp16_64_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 25.6kB/s]
100%|█████████████████████████████████████████████████████████████████████████████████| 640/640 [00:00<00:00, 27.8kB/s]
No vmfb found. Compiling and saving to H:\branches\shark\web\clip_19dec_v2p1base_fp32_64_vulkan-00000000-0100-0000-0000-000000000000.vmfb
Using target triple -iree-vulkan-target-triple=rdna2-unknown-windows from command line args
Saved vmfb in H:\branches\shark\web\clip_19dec_v2p1base_fp32_64_vulkan-00000000-0100-0000-0000-000000000000.vmfb.
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Traceback (most recent call last):
  File "H:\branches\shark\shark.venv\lib\site-packages\gradio\routes.py", line 322, in run_predict
    output = await app.get_blocks().process_api(
  File "H:\branches\shark\shark.venv\lib\site-packages\gradio\blocks.py", line 1015, in process_api
    result = await self.call_function(
  File "H:\branches\shark\shark.venv\lib\site-packages\gradio\blocks.py", line 833, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "H:\branches\shark\shark.venv\lib\site-packages\anyio\to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "H:\branches\shark\shark.venv\lib\site-packages\anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "H:\branches\shark\shark.venv\lib\site-packages\anyio\_backends\_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "H:\branches\shark\web\models\stable_diffusion\img2img_main.py", line 134, in img2img_inf
    latents = vae_encode("forward", (init_img,))
  File "H:\branches\SHARK\shark\shark_inference.py", line 136, in __call__
    return self.shark_runner.run(function_name, inputs, send_to_host)
  File "H:\branches\SHARK\shark\shark_runner.py", line 90, in run
    return get_results(
  File "H:\branches\SHARK\shark\iree_utils\compile_utils.py", line 362, in get_results
    result = compiled_vm[function_name](*device_inputs)
  File "H:\branches\shark\shark.venv\lib\site-packages\iree\runtime\function.py", line 130, in __call__
    self._invoke(arg_list, ret_list)
  File "H:\branches\shark\shark.venv\lib\site-packages\iree\runtime\function.py", line 154, in _invoke
    self._vm_context.invoke(self._vm_function, arg_list, ret_list)
IndexError: Error invoking function: D:\a\SHARK-Runtime\SHARK-Runtime\c\runtime\src\iree\modules\hal\module.c:1118: OUT_OF_RANGE; count 102 of iree_hal_fence > 32; while invoking native function hal.fence.join; while calling import;
[ 1]   native hal.fence.join:0 -
[ 0] bytecode module.forward:4384 /home/jinchen/SHARK/shark.venv/lib/python3.10/site-packages/torch/_ops.py:499:0

**Comment by fokin33** [comment]
No error now, but i always got black picture in UI after generation img2img. txt2img work correctly.

**Comment by fokin33** [comment]
If select any other model than  stablediffusion i got error again

Found device AMD Radeon RX 580 2048SP. Using target triple rdna2-unknown-windows.
Tuned models are currently not supported for this setting.
Using cached models from h:\shark\TEMP...
No internet connection. Using the model already present in the tank.
loading existing vmfb from: H:\branches\shark\web\euler_scale_model_input_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Using cached models from h:\shark\TEMP...
No internet connection. Using the model already present in the tank.
loading existing vmfb from: H:\branches\shark\web\euler_step_fp16_vulkan-00000000-0100-0000-0000-000000000000.vmfb
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Traceback (most recent call last):
  File "H:\branches\shark\web\models\stable_diffusion\opt_params.py", line 31, in get_params
    model_name = models_db[1][model_key]
KeyError: 'anythingv3/v2_1base/vae_encode/fp16/length_77/untuned'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "H:\branches\shark\shark.venv\lib\site-packages\gradio\routes.py", line 322, in run_predict
    output = await app.get_blocks().process_api(
  File "H:\branches\shark\shark.venv\lib\site-packages\gradio\blocks.py", line 1015, in process_api
    result = await self.call_function(
  File "H:\branches\shark\shark.venv\lib\site-packages\gradio\blocks.py", line 833, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "H:\branches\shark\shark.venv\lib\site-packages\anyio\to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "H:\branches\shark\shark.venv\lib\site-packages\anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "H:\branches\shark\shark.venv\lib\site-packages\anyio\_backends\_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "H:\branches\shark\web\models\stable_diffusion\img2img_main.py", line 111, in img2img_inf
    model_cache.set_models(device_key)
  File "H:\branches\shark\web\models\stable_diffusion\cache_objects.py", line 111, in set_models
    self.vae_encode = get_vae_encode()
  File "H:\branches\shark\web\models\stable_diffusion\opt_params.py", line 81, in get_vae_encode
    bucket, model_name, iree_flags = get_params(
  File "H:\branches\shark\web\models\stable_diffusion\opt_params.py", line 36, in get_params
    raise Exception(
Exception:  there is no entry for anythingv3/v2_1base/vae_encode/fp16/length_77/untuned in the models database

**Pull request closed without merging**